app/views/templates/_form.html.erb

```diff
@@ -43,16 +43,25 @@
               <input type="radio" name="options" id="option1" ><%= _("Code") %>
             </label>
             <label class="btn btn-default btn-sm" onclick="set_preview()">
-              <input type="radio" name="options" id="option2" ><%= _("Preview") %>
+              <input type="radio" name="options" id="option2" ><%= _("Diff") %>
+            </label>
+            <label class="btn btn-default btn-sm" onclick="set_render()">
+              <input type="radio" name="options" id="option3" ><%= _("Preview") %>
             </label>
           </div>
+          <span id="preview_host_selector" style="display:none">
+            <%= select_tag :preview_host_id, options_from_collection_for_select(Host.authorized(:view_hosts).limit(100), :id, :name),
```

Comment:
Perhaps also search for hosts that are running an associated OS, so the list is more accurate